src/oom/oomd-util.c

```diff
@@ -251,10 +263,18 @@ int oomd_kill_by_swap_usage(Hashmap *h, bool dry_run) {
                         continue;
 
                 r = oomd_cgroup_kill(sorted[i]->path, true, dry_run);
-                if (r > 0 || r == -ENOMEM)
+                if (r > 0) {
+                        selected = strdup(sorted[i]->path);
+                        if (!selected)
+                                return -ENOMEM;
+                        break;
+                }
+                if (r == -ENOMEM)
                         break;
         }
 
+        if (selected)
+                *ret_selected = TAKE_PTR(selected);
```

Comment:
This doesn't matter too much, since this is only called from a few places, but in general we try to initialize output variables on success, and not initialize them on failure. And `selected` is set once and unconditionally used below. So maybe simlify this a bit:
```C
   r = oomd_cgroup_kill(sorted[i]->path, true, dry_run);
   if (r == -ENOMEM)
         return r;
   if (r <= 0) /* smart comment here */
         continue;

    char *selected = strdup(sorted[i]->path);
    if (!selected)
           return -ENOMEM;
   *ret_selected = selected;
   return r;
```

Also, this currently might return an error if oomd_cgroup_kill failed, but without any logging. This seems sloppy. I think there should be a variable like `ret = 0`, and then on failure do
```
if (ret >= 0 && r < 0)
      ret = r;
```
So that the function can return the first failure, if any.